packages/core/src/events/CoreEventHandlers.ts

```diff
@@ -13,7 +13,11 @@ export class CoreEventHandlers<O = {}> extends EventHandlers<
       hover: (el: HTMLElement, id: NodeId) => {},
       drag: (el: HTMLElement, id: NodeId) => {},
       drop: (el: HTMLElement, id: NodeId) => {},
-      create: (el: HTMLElement, UserElement: React.ReactElement) => {},
+      create: (
+        el: HTMLElement,
+        UserElement: React.ReactElement,
+        callbackFunction?: (nodeTree: NodeTree) => void
```

Comment:
In this case, the required value will be the `UserElement`; where as the 3rd parameter will be the `options` object. For example:

```suggestion
      create: (
        el: HTMLElement,
        UserElement: React.ReactElement,
        options?: { onCreate: (nodeTree: NodeTree) => void }
```

Reply:
So with #152 in mind, options should be typed:
```
interface OnCreateOptions {
  onCreate?: (nodeTree: NodeTree) => void;
  previewImage?: string;
}
```

Reply:
Can we do `onCreate` first? Then we can redo the `previewImage` feature in the next PR? Just so we can keep things atomic and easily reversible 😅

Reply:
Yes! That was my plan, but poorly communicated on my part